client/Bootstrap/Navigation/README.md

```diff
@@ -0,0 +1,271 @@
+
+To achieve this, the component accepts [the page configuration](../../Pages/README.md) via the `configuration` property, [which is then processed](#route-configuration-processing), resulting in a [React Router](https://github.com/ReactTraining/react-router) based routing implementation derived from that configuration. React Router will handle URL parsing, and will render a given component when a particular route is accessed by the user.
+
+In addition to this, the Navigation component is also responsible for rendering the visual Navigation components that an end user interacts with, such as the 1st, 2nd or 3rd levels of navigation. It renders these as needed, based on the current page a user is viewing, and the [type of page it defines itself as](#page-types). These page types are declarative to encapsulate the different types of Navigation visual components used, and are defined and exported as a part of this component.
+
+The final responsibility this component is to handle asynchronous loading of pages via React Lazy and Suspense (if required), and to provide the pages with access to helper functions and navigation state, such as path parameters or redirection functions to other pages.
+
+## Properties
+
+The properties expected by this component are as follows:
+
+| Property        | Type                                                                                            | Required | Default value (if no value provided)                                                          |
+| --------------- | ----------------------------------------------------------------------------------------------- | -------- | --------------------------------------------------------------------------------------------- |
+| `configuration` | Object, values of which implement the [page schema](../../Pages/README.md#page-metadata-schema) | No       | The exported content from the `Pages` module (ie all pages from the `Client/Pages` directory) |
+
+Additional properties to follow in a later PR.
+
+## Example usage
+
+To follow in a later PR.
+
+_Note/To do_: Describe/define how path parameters, as well as helper functions, such as 'go back to the previous Strimzi ui page', 'go to this page by ID', or 'is this page available by ID', are passed to pages mounted by the router.
+
+## Route configuration processing
+
+As mentioned, this component takes in [configuration](../../Pages/README.md) which maps [`Panels`](../../Panels/README.md) to routes, along with other supporting metadata, such as a translatable page names, what backend services are required to show the page, and what authorization(s) a user may need to see it. This configuration is processed via the Navigation's model, and will result in an object which is then iterated on in the Navigation's view layers to generate the UI's routing/navigation. For example, given the following configuration:
```

Comment:
`home/topic/1/partitions` - what if that was to render the topic page, but with a slide out panel enabled?

Might be me thinking too hard about this - but it looks like this is tying us such that a path maps to a single component.

And is there a way to add dynamic properties (i.e pass through something like a query param or a route param) - so that the component is unaware of the history or route objects?

Reply:
For elements like that, I would consider a query parameter perhaps. The job of the Navigation is to render routes based on configuration, and pass contextual information (path parameters, query parameters) to the configured component. What the component does with those pieces of information are not in my opinion the responsibility of the navigation component.

To your second point, I have prototyped and validated the ability to pass things like path paramters, query parameters via a property to the component on render. I did leave it as a todo for implementation, but I will update the PR with more details on this now.